src/reconcile/rec_write.c

```diff
@@ -2895,10 +2895,13 @@ __wt_bulk_init(WT_SESSION_IMPL *session, WT_CURSOR_BULK *cbulk)
 		WT_RET_MSG(session, EINVAL,
 		    "bulk-load is only possible for newly created trees");
 
-	/* Set a reference to the empty leaf page. */
+	/* Create an empty leaf page to insert into. */
 	pindex = WT_INTL_INDEX_COPY(btree->root.page);
 	cbulk->ref = pindex->index[0];
+	WT_RET(__wt_cache_read(session, cbulk->ref));
 	cbulk->leaf = cbulk->ref->page;
+	WT_RET(__wt_page_modify_init(session, cbulk->leaf));
+	__wt_page_only_modify_set(session, cbulk->leaf);
```

Comment:
@michaelcahill, I'm worried about error handling here -- if we fail after the page allocation, don't we have to discard the ref/page combination?

I'm also inclined to change the call to `__wt_cache_read` to simply allocate the page, we use almost none of `__wt_cache_read` -- why did you do it that way?